wpilibj/src/athena/java/edu/wpi/first/wpilibj/QuantityEncoder.java

```diff
@@ -0,0 +1,80 @@
+package edu.wpi.first.wpilibj;
+
+import tec.uom.se.quantity.Quantities;
+
+import javax.measure.Quantity;
+import javax.measure.Unit;
+
+/**
+ *
+ * @param <Q> The Quantity being measured. (ex: Length, Angle)
+ * @param <S> The Speed of the Quantity being measured. (ex: Speed, AngularSpeed)
+ */
+// Package private so it is not part of the public API
+abstract class QuantityEncoder<Q extends Quantity<Q>, S extends Quantity<S>> {
+
+  private final Encoder m_encoder;
```

Comment:
Sure, I imagine we could provide a better user experience by deprecating Encoder. However, Encoder is probably one of the most popular, well-known classes in the library - deprecating it will have a cost to the community, in terms of updating documentation, people fixing code, forum posts becoming outdated, etc.

How much value do you see in the addition of units here? Does it really outweigh that cost?

Reply:
(With regards to distancePerPulse, perhaps it could change the assigned unit, say to Dimenless by default? We might also deprecate that method in favor of one that takes a unit as a second argument )

Reply:
*Dimensionless